src/main/cpp/rollingfileappender.cpp

```diff
@@ -340,11 +340,13 @@ bool RollingFileAppenderSkeleton::rollover(Pool& p)
 						}
 						else
 						{
+							closeWriter();
```

Comment:
[activateOptions](https://github.com/apache/logging-log4cxx/blob/9cf215f9721aa6add98674bfc706452e62268b50/src/main/cpp/fileappender.cpp#L155) calls [setFile](https://github.com/apache/logging-log4cxx/blob/9cf215f9721aa6add98674bfc706452e62268b50/src/main/cpp/fileappender.cpp#L164) as well which calls [closeWriter](https://github.com/apache/logging-log4cxx/blob/9cf215f9721aa6add98674bfc706452e62268b50/src/main/cpp/fileappender.cpp#L281) already. So the explicit call might not be necessary anymore.

Though, just a hint, not worth blocking the PR.